content/6_Advanced/DP_More.mdx

```diff
@@ -105,16 +114,99 @@ Queue with 2 stacks is used to remove a factor of $\mathcal{O}(\log N)$ in [USAC
 
 ## DP on Broken Profile
 
-<Resources>
+<!-- <Resources>
 	<Resource
 		source="cp-algo"
 		title="DP on Broken Profile"
 		url="dynamic_programming/profile-dynamics.html"
 	>
 		{' '}
 	</Resource>
-</Resources>
+</Resources> -->
 
-<IncompleteSection />
+Broken profile DP is a subset of bitmask DP. Problems falling under this category generally have the following properties:
+1. They're about filling a 2D grid.
+2. One of the dimensions is much smaller than the other.
+3. When filling the grid, each cell depends only on adjacent cells.
+4. The cells don't have many possible values (usually only 2).
+
+The third property is especially important, as it means that we can process the cells column-by-column (imagine a snake wrapping around the grid). We then only need to care about the rightmost processed cell in each row (hence the name "broken profile").
+
+The fourth property suggests that we should use a bitmask to represent that broken profile.
+
+<Warning>
+
+Note that the bitmask doesn't necessarily have to represent the state of the *cells*. In some problems (e.g. CEOI 2006 Connect), it can represent the state of the cell *borders* instead.
+
+</Warning>
+
+<FocusProblem problem={problems.brokenFocus} />
+
+### Tutorial
+
+Let $\texttt{dp}[i][j][mask]$ be the number of ways to tile the grid so that:
+- All cells from cell $(0, 0)$ to cell $(i - 1, j)$ are **covered**.
+- All cells from cell $(i, j + 1)$ to cell $(N, M)$ are **empty**.
+- $mask$ represents whether each of the remaining $N$ cells are **empty**, with the $k$-th bit corresponding to the cell in column $k$.
+
+For example, the following state would contribute toward $\texttt{dp}[3][1][10100_2]$:
+![](broken-profile-state-example.png)
+
+We now have two cases when calculating $\texttt{dp}[i][j][mask]$:
+- The $j$-th bit of mask is 0, meaning that cell $(i, j)$ is **covered**.
+	- Case 1: we used a vertical tile to cover it.
+		- Cell $(i - 1, j)$ must have been empty before this, so there are $\texttt{dp}[i][j - 1][mask \oplus 2^j]$ ways to do this.
+	- Case 2: we used a horizontal tile to cover it.
+		- This is only possible if $j > 0$.
+		- Cells $(i, j - 1)$, $(i - 1, j - 1)$, and $(i - 1, j)$ must have been covered, so there are $\texttt{dp}[i][j - 2][mask]$ to do this.
+- The $j$-th bit of the mask is 1, meaning that cell $(i, j)$ is **empty**.
+	- Cell $(i - 1, j)$ must have been covered before this, so there are $\texttt{dp}[i][j - 1][mask \oplus 2^j]$ ways to do this.
+
+Note that the indices we need to use may become negative and will thus require wrapping. To simplify calculations and bypass this, simply drop
```

Comment:
you can write it so that you only need to keep two DP arrays at a time, not three